### PR TITLE
BMI270 driver

### PR DESCRIPTION
**Pull request by andyp1per**
Driver for the BMI270 and update to the BeastF7v2 which uses it

### CODE REVIEW

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp [hunk 1]
```diff
@@ -0,0 +1,573 @@
+/*
+ * Some parts Copyright (C) 2016  Intel Corporation. All rights reserved.
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:2]: remove copy-pasted intel copyright

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+* COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+* STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+* IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+static const uint8_t bmi270_maximum_fifo_config_file[] = { BMI270_REG_INIT_DATA,
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:108]: this should be a static const class member

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.h [hunk 3]
```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2016  Intel Corporation. All rights reserved.
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI270.h:2]: remove intel copyright

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.h [hunk 4]
```diff
@@ -123,6 +123,7 @@ class AP_InertialSensor_Backend
         DEVTYPE_INS_ICM42605 = 0x35,
         DEVTYPE_INS_ICM40605 = 0x36,
         DEVTYPE_INS_IIM42652 = 0x37,
+        DEVTYPE_BMI270       = 0x38,
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_Backend.h:126]: need to add to Tools/scripts/decode_devid.py

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+    }
+
+    return sensor;
+}
+
+void AP_InertialSensor_BMI270::start()
+{
+    _dev->get_semaphore()->take_blocking();
+
+    if (!_configure_accel()) {
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:208]: make the 3 configure fns void and removed the panic calls, it just wastes flash space, as they always return true

**Review by tridge — CHANGES_REQUESTED** [review]
congratulations on getting this working!

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp [hunk 6]
```diff
@@ -0,0 +1,573 @@
...
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+
+#include <AP_HAL/utility/sparse-endian.h>
+#include <AP_HAL_Linux/GPIO.h>
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:22]: remove HAL_Linux include

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp [hunk 7]
```diff
@@ -0,0 +1,660 @@
...
+* COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+* STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+* IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+const uint8_t bmi270_maximum_fifo_config_file[] = {
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp]: Fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:2]: Fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:108]: Fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.h (hunk 3 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI270.h:2]: Fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp (hunk 5 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:208]: Fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp (hunk 6 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_BMI270.cpp:22]: Fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.h (hunk 4 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.h:126]: Fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**